access/email/plugindata.go

```diff
@@ -0,0 +1,98 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/teleport/api/types"
+)
+
+// PluginData is a data associated with access request that we store in Teleport using UpdatePluginData API.
+type PluginData struct {
+	RequestData
+	EmailThreads []EmailThread
+}
+
+// Resolution is a review request resolution result
+type Resolution struct {
+	Tag    ResolutionTag
+	Reason string
+}
+type ResolutionTag string
+
+const Unresolved = ResolutionTag("")
+const ResolvedApproved = ResolutionTag("APPROVED")
+const ResolvedDenied = ResolutionTag("DENIED")
+const ResolvedExpired = ResolutionTag("EXPIRED")
+
+// RequestData represents part of plugin data responsible for review request
+type RequestData struct {
+	User          string
+	Roles         []string
+	RequestReason string
+	ReviewsCount  int
+	Resolution    Resolution
+}
+
+// EmailThread stores value about particular original message
+type EmailThread struct {
+	Email     string
+	MessageID string
+	Timestamp string
+}
+
+// NewRequestData converts types.AccessRequest to RequestData
+func NewRequestData(req types.AccessRequest) (string, RequestData) {
+	id := req.GetName()
```

Comment:
The caller of `NewRequestData` can easily `req.GetName()` too cause it's a trivial getter.
Why we do it here? It seems like an alien thing to `RequestData` construction.

Reply:
Yes. I do not see it as an alien thing. ID belongs to request data, except that for some reason it is not embedded into struct and is always a separate variable.